### DISCUSSION TITLE
CUDA locks GPU in P2 state

### DISCUSSION THREAD

**Discussion by thesword53** [NVIDIA/open-gpu-kernel-modules#261] [Ideas]
Every software that creates a CUDA context locks GPU at P2 state until the context is destroyed even with NVENC/NVDEC. It forces the GPU clock to stay at the maximum and decreases memory clock. There is no way to disable this driver behavior and it happens on both Linux and Windows driver. On Windows P2 state can be disabled but the GPU stays at P0 state.

The consequences are:

- More GPU power consumption, no power saving and less battery duration for laptops especially with NVDEC video decode.
- Slightly less performance in gaming when users streams or record games with NVENC encoder.

There are some discussions opened about it:

- https://forums.developer.nvidia.com/t/nvdec-forces-gpu-into-p2-cuda-state-much-higher-power-consumption-than-with-vdpau/55466
- https://forums.developer.nvidia.com/t/whats-the-situation-with-vdpau-vaapi-nvdec/61031/10
- https://www.reddit.com/r/nvidia/comments/aco4ks/how_to_turn_off_nvidias_force_p2_power_state/

**Comment by 4budab1** [comment]
i disabled it via profile inspector during mining age,
now i disable it for using during CUDA programming =)
for what reason it even introduced?

**Comment by wgpierce** [comment]
This is fixed on the latest R580 and all future branches:
https://forums.developer.nvidia.com/t/whats-the-situation-with-vdpau-vaapi-nvdec/61031/51
